js/url.ts

```diff
@@ -64,7 +64,7 @@ function parse(url: string): URLParts | undefined {
 }
 
 export class URL {
-  private _parts: URLParts;
+  _parts: URLParts;
```

Comment:
Why remove private ?

Reply:
in spec, `URLSearchParams`

> 1. A URLSearchParams object has an associated list of name-value pairs, which is initially empty.
> 1. A URLSearchParams object **has an associated url object**, which is initially null.

and when delete a key, run a [update steps](https://url.spec.whatwg.org/#concept-urlsearchparams-update):

> A `URLSearchParams` object’s **update steps** are to run these steps:
> 
>     1. Let query be the serialization of URLSearchParams object’s list.
>     1. If query is the empty string, then set query to null.
>     1. **Set url object’s url’s query to query**.

In the current code implementation, the `_parts` is "url object’s url", if it is private, we can't access its `query`.

Reply:
I also considered another way (trick), when access `_parts` outside the class `URLSearchParams`, convert it to `any` that has avoided the private limit.

```ts
(this.url as any)._parts.query = query;
```

Reply:
I would use the `any` hammer over changing the public API surface of an object.  Though it seems like that is a really odd breaking of encapsulation in the standard.  How does JSDom accomplish this?

Reply:
jsdom use [webidl2js](https://github.com/jsdom/webidl2js) to generate a "wrapper class" `URL`, delegating to an "implementation class". the `_url`  only exists  on impls, not on wrappers.